spec/unit/provider/package/dnfmodule_spec.rb

```diff
@@ -202,12 +202,45 @@
     end
   end
 
+  describe "when disabling a module" do
+
+    let(:name) { 'baz' }
+
+    let(:resource) do
+      Puppet::Type.type(:package).new(
+        :name => name,
+        :provider => 'dnfmodule',
+      )
+    end
+
+    let(:provider) do
+      provider = described_class.new
+      provider.resource = resource
+      provider
+    end
+
+    it "executed the disable command" do
+      resource[:ensure] = :disabled
+      expect(provider).to receive(:execute).with(array_including('disable'))
+      provider.disable
+    end
+
+    it "does not try to disable if package is already disabled" do
+      allow(described_class).to receive(:command).with(:dnf).and_return(dnf_path)
+      allow(Puppet::Util::Execution).to receive(:execute)
+        .with("/usr/bin/dnf module list --enabled -d 0 -e 1;/usr/bin/dnf module list --disabled -d 0 -e 1")
+        .and_return("baz 1.2 [d][x] common [d], complete  Package Description")
+      resource[:ensure] = :disabled
+      expect(provider).to be_insync(:disabled)
+    end
+  end
+
   context "parsing the output of module list --enabled" do
     before { allow(described_class).to receive(:command).with(:dnf).and_return(dnf_path) }
 
     it "returns an array of enabled modules" do
       allow(Puppet::Util::Execution).to receive(:execute)
-        .with("/usr/bin/dnf module list --enabled -d 0 -e 1")
+        .with("/usr/bin/dnf module list --enabled -d 0 -e 1;/usr/bin/dnf module list --disabled -d 0 -e 1")
         .and_return(packages)
 
       enabled_packages = described_class.instances.map { |package| package.properties }
```

Comment:
It would be good to rename and update the fixture here: https://github.com/puppetlabs/puppet/blob/master/spec/fixtures/unit/provider/package/dnfmodule/dnf-module-list-enabled.txt to also add some disabled and default (neither enabled nor disabled) modules.